pkg/cmd/cluster/config_resource.go

```diff
@@ -271,3 +343,76 @@ func (w *configObjectsWrapper) genConfigSpecs(objects *ConfigRelatedObjects, com
 	}
 	return ret, nil
 }
+
+func (w *configObjectsWrapper) genConfigSpecsByCompDef(comp *appsv1alpha1.Component, compDefs []*appsv1alpha1.ComponentDefinition) ([]*configSpecMeta, error) {
+	var (
+		ret []*configSpecMeta
+	)
+	if comp == nil {
+		return nil, fmt.Errorf("failed to get the cluster component CR in K8s")
```

Comment:
this error is confusing, under what circumstances would 'comp' be nil?

Reply:
In this PR, I can ensure that the `comp` field, which enters this method, is not nil. However, I am concerned that other people may not perform the necessary checks before calling this function, and since the subsequent code also uses pointer types, I have added an early checking logic.